packages/uploadthing/src/internal/handler.ts

```diff
@@ -282,6 +289,9 @@ export const buildRequestHandler = <TRouter extends FileRouter>(
           });
         }
 
+        const url = new URL(input.url ?? req.url ?? "");
```

Comment:
if this falls all the way through to the `""`, it will likely throw as an invalid URL

Reply:
yea :/ this fallback was there before though so i didn't do anything about it.

we could wrap in try-catch and throw a better error maybe?

Reply:
Is it even possible for both `input.url` and `req.url` to be empty?  I am trying to figure out what case falls though in the first place. 

Regardless, try catch and a better error is probably best.

Reply:
not really... the input.url is provided by us for adapters that doesn't provide an absolute url the req.url - so if the "" is reached we messed up

Reply:
Does TS get mad if the `""` is not there?

Reply:
yes

Reply:
I guess just 

```
new URL((input.url ?? req.url) as string)
```

and wrap in a try catch just in case

Reply:
done https://github.com/pingdotgg/uploadthing/pull/310/commits/eb2119fea654f89cb55d975faeb5e6c691037e7f